src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala

```diff
@@ -0,0 +1,50 @@
+package org.broadinstitute.dsde.rawls.dataaccess
+
+import com.tinkerpop.blueprints.{Graph, Vertex}
+import com.tinkerpop.gremlin.java.GremlinPipeline
+import com.tinkerpop.pipes.PipeFunction
+
+import scala.collection.JavaConversions._
+
+trait GraphDAO {
+  val db: Graph
+
+  def hasProperty[T](key: String, value: T) = new PipeFunction[Vertex, java.lang.Boolean] {
+    override def compute(v: Vertex) = v.getProperty[T](key).equals(value)
+  }
+
+  // TODO be able to understand different types?
+  def hasProperties(props: Map[String, Object]) = new PipeFunction[Vertex, java.lang.Boolean] {
+    override def compute(v: Vertex) = props.map(p => v.getProperty(p._1).equals(p._2)).reduce(_&&_)
+  }
+
+  implicit def toPipeFunction[A, B](f: A => B) = new PipeFunction[A, B] {
+    override def compute(a: A): B = f(a)
+  }
+
+  def isReservedProperty(key: String) = key.startsWith("_")
+
+  def getWorkspaceVertex(namespace: String, name: String): Option[Vertex] = {
```

Comment:
it does not seem right to have getWorkspaceVertex and getEntityVertex here. Perhaps instead we have a catalog of filter functions... essentially named queries